packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts

```diff
@@ -231,6 +289,74 @@ export class CfnInclude extends core.CfnElement {
     this.conditions[conditionName] = cfnCondition;
   }
 
+  private createNestedStack(nestedStackId: string, props?: CfnIncludeProps): void {
+    const resources = this.template.Resources;
+
+    if (!(nestedStackId in resources)) {
+      throw new Error(`Nested Stack with logical ID '${nestedStackId}' was not found in the template`);
+    }
+
+    if (!(resources[nestedStackId].Type === 'AWS::CloudFormation::Stack' && props?.nestedStacks)) {
+      throw new Error(`Nested Stack with logical ID '${nestedStackId}' is not an AWS::CloudFormation::Stack resource`);
+    }
+
+    const self = this;
+    const cfnParser = new cfn_parse.CfnParser({
+      finder: {
+        findCondition(conditionName: string): core.CfnCondition | undefined {
+          return self.conditions[conditionName];
+        },
+
+        findResource(lId: string): core.CfnResource | undefined {
+          if (!(lId in (self.template.Resources || {}))) {
```

Comment:
Hmmm. This cannot be correct. This means nested stack cannot depend on each other. For example, I believe this will fail today:

```json
{
  "Resources": {
    "ChildStack1": {
      "Type": "AWS::CloudFormation::Stack",
      "DependsOn": "ChildStack2"
    },
    "ChildStack2": {
      "Type": "AWS::CloudFormation::Stack"
    }
  }
}
```

I think the problem here is that you have a separate loop with creating nested stacks. If you moved creating nested stacks into the `getOrCreateResource()` method, I believe this problem will go away.